src/spatial/topology.jl

```diff
@@ -61,6 +69,24 @@ function neighbors(grid, site)
     Iterators.map(off -> LI[off+I], Iterators.filter(off -> off+I in CI, grid.offsets))
 end
 
+"""
+given a vector of hopping constants of length num_nbs(grid, site), return a vector of size 2*(dimension of grid) with zeros at indices where the neighbor is out of bounds
+"""
+function pad_hop_vec(grid, site, hop_vec::Vector{F}) where F <: Number
+    length(hop_vec) == 2*dimension(grid) && return hop_vec
+    new_hop_vec = zeros(2*dimension(grid))
+    CI = grid.CI; offsets = grid.offsets
+    @inbounds I = CI[site]
+    nbr_counter = 1
+    for (i,off) in enumerate(grid.offsets)
+        if I + off in CI
+            new_hop_vec[i] = hop_vec[nbr_counter]
```

Comment:
@inbounds

Reply:
This is in a constructor, which will only be called once at aggregation step. Do we really need `@inbounds` for things that are called once?